### ISSUE TITLE
Early stopping

### ISSUE THREAD

**Issue by polars05** [edwardzhou130/PolarSeg#3]
Does ```train.py``` do early stopping as mentioned in ```README.md```? I can't seem to find the relevant section of code in ```train.py``` where early stopping has been implemented.

**Comment by YangZhang4065** [comment]
Hi @polars05 , 

Sorry for the confusion. The model will be trained indefinitely until manually terminated so that you won't miss any possible validation improvement. Another reason is the validation curves and convergence criteria vary in different hyperparameters and computational resources.

The model is saved whenever there is a higher validation mIoU anyway. You can consider it to be an early stopping with a manually determined grace period.

**Comment by polars05** [comment]
Thanks for the clarifications, @YangZhang4065! Out of curiosity, how many epochs did you train for and what criteria did you consider in choosing whether to (manually) terminate training?

**Comment by edwardzhou130** [comment]
In our experiments, PolarNet usually reached the best mIoU performance on the validation set within 20 epochs.
